tests/test_coastal_vulnerability.py

```diff
@@ -1267,6 +1267,49 @@ def count_shore_points(landmass_geometries):
 
         self.assertEqual(n_points_expected, n_points_actual)
 
+    def test_landmass_with_redundant_vertices(self):
+        """CV: test landmass processing with redundant landmass vertices."""
+        workspace_dir = self.workspace_dir
+
+        aoi_path = os.path.join(workspace_dir, 'aoi.geojson')
+        landmass_path = os.path.join(workspace_dir, 'landmass.geojson')
+        polygon_pickle = os.path.join(
+            workspace_dir, 'polygon.pickle')
+        lines_pickle = os.path.join(
+            workspace_dir, 'lines.pickle')
+        lines_rtree = os.path.join(
+            workspace_dir, 'rtree.dat')
+        target_vector_path = os.path.join(
+            workspace_dir, 'shore_points.gpkg')
+        model_resolution = 1
+        srs = osr.SpatialReference()
+        srs.ImportFromEPSG(32634)  # UTM Zone 34N
+        wkt = srs.ExportToWkt()
+
+        # Nearly identical vertices are from real data that raised an error
+        landmass_coords = [
+            (420274.0, 6042938.0),
+            (420275.7984616568, 6042939.285698998),  # nearly identical
+            (420275.7984616570, 6042939.285698998),  # nearly identical
+            (420276.0, 6042938.0),
+            (420274.0, 6042938.0)]

```

Comment:
This is a really nicely constructed bit of sample data, and I particularly like the `# nearly identical` highlight to emphasize what is relevant to the test.